config/routes.rb

```diff
@@ -0,0 +1,75 @@
+Rails.application.routes.draw do
+  # The priority is based upon order of creation: first created -> highest priority.
+  # See how all your routes lay out with "rake routes".
+  root 'markets#index'
+
+  get "products/new/:id" => "products#new", as: "new_product"
+
+  get "sales/new/:id" => "sales#new", as: "new_sale"
+
+  get "vendors/new/:id" => "vendors#new", as: "new_vendor"
+
+  get "sales/:id/:name" => "sales#month_sale", as: "month_sale"
+
+  get "markets/1/:letter" => "markets#lettermarket", as: "letter_market"
```

Comment:
This seems like an unusual way to create the route. It might have been better to do "markets/:letter" or "markets/search/:letter".  Having the "1" is confusing and doesn't follow traditional routing practices.